File: app/src/main/java/com/tevinjeffrey/rutgersct/ui/sectioninfo/SectionInfoSubcomponent.java
package com.tevinjeffrey.rutgersct.ui.sectioninfo;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;

@Subcomponent
public interface SectionInfoSubcomponent
    extends AndroidInjector<SectionInfoFragment> {

  void inject(SectionInfoPresenterImpl sectionInfoPresenter);

  @Subcomponent.Builder
  abstract class Builder extends AndroidInjector.Builder<SectionInfoFragment> {
    @Override public abstract SectionInfoSubcomponent build();
  }
}

File: app/src/main/java/com/tevinjeffrey/rutgersct/ui/sectioninfo/SectionInfoSubcomponent.kt
package com.tevinjeffrey.rutgersct.ui.sectioninfo

import dagger.Subcomponent
import dagger.android.AndroidInjector

@Subcomponent
interface SectionInfoSubcomponent : AndroidInjector<SectionInfoFragment> {

  fun inject(sectionInfoViewModel: SectionInfoViewModel)

  @Subcomponent.Builder
  abstract class Builder : AndroidInjector.Builder<SectionInfoFragment>() {
    abstract override fun build(): SectionInfoSubcomponent
  }
}
